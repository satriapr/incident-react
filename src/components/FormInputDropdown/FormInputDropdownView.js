import React from 'react';
import { MenuItem, TextField } from '@mui/material';
import { Controller } from 'react-hook-form';
import PropTypes from 'prop-types';
import { get } from 'lodash';

// Common form input dropdown. React-hook-form
const FormInputDropdown = ({ name, control, label, options }) => {
  const generateSelectOptions = () => {
    return options.map((option) => {
      return (
        <MenuItem
          key={get(option, 'value', '')}
          value={get(option, 'value', '')}
        >
          {option.label}
        </MenuItem>
      );
    });
  };

  return (
    <Controller
      control={control}
      name={name}
      rules={{ required: `${label} is required` }}
      render={({ field: { onChange, value }, fieldState: { error } }) => (
        <TextField
          onChange={onChange}
          value={value || value === 0 ? value : ''}
          error={!!error}
          helperText={error ? error.message : null}
          select
          label={label}
          required
          fullWidth
        >
          {generateSelectOptions()}
        </TextField>
      )}
    />
  );
};

export default FormInputDropdown;

FormInputDropdown.defaultProps = {
  name: '',
  control: {},
  label: '',
  options: []
};

FormInputDropdown.propTypes = {
  name: PropTypes.string,
  control: PropTypes.objectOf(PropTypes.any),
  label: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.object)
};
