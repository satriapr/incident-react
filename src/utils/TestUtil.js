import { shallow } from 'enzyme';
import * as React from 'react';

const singleSnapTest = (wrapper, description) => {
  test(description, () => {
    expect(wrapper).toMatchSnapshot();
  });
};

const testSnapshots = (Component, configs) => {
  describe('snapshots', () => {
    configs.forEach((config) => {
      const { props, description, state } = config;
      const wrapper = shallow(<Component {...props} />);
      if (state) wrapper.setState(state);
      singleSnapTest(wrapper, description);
    });
  });
};

export { testSnapshots };
