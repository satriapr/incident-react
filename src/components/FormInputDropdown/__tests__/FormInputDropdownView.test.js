import FormInputDropdown from '../FormInputDropdownView';
import { testSnapshots } from '../../../utils/TestUtil';

describe('FormInputDropdown Component', () => {
  testSnapshots(FormInputDropdown, [
    {
      description: 'FormInputDropdown props',
      props: {
        name: '',
        control: {},
        label: '',
        options: []
      }
    }
  ]);
});
