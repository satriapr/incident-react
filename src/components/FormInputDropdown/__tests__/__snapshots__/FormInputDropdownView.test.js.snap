// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputDropdown Component snapshots FormInputDropdown props 1`] = `ShallowWrapper {}`;
